runtimes/pythonrt/ak_runner/deterministic.py

```diff
@@ -10,9 +10,33 @@
 
 def is_determinstic(fn):
     """Return True if fn (callable) can run outside of activity."""
-    return fn in functions or fn.__module__ in modules
+    if fn in functions:
+        return True
 
-# Please keep the following modules and functions sorted in alphabetical order.
+    if fn.__module__ in modules:
+        return True
+
+    self = getattr(fn, '__self__', None)
+    if self is not None:
```

Comment:
There's no values of `__self__` that can be evaluated as `False` other than `None`, right?
```suggestion
    if self:
```

Reply:
Not sure and I don't want to risk it. Since I default to `None` I check for it.